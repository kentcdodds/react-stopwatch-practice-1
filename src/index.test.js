import React from 'react'
import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import StopWatch from './index'

jest.useFakeTimers()

test('initial state', () => {
  expect(mountToJson(mount(<StopWatch buttonStyles={{fontSize: '2em'}} />))).toMatchSnapshot()
})

test('full interaction', () => {
  const wrapper = mount(<StopWatch buttonStyles={{fontSize: '2em'}} />)
  const toggle = getToggle(wrapper)
  // initial state
  expect(toggle.text()).toBe('Start')
  expect(getMillisecondsInt(wrapper)).toBe(0)

  const fastForward = 10
  clickToggle(wrapper)
  jest.runTimersToTime(fastForward)

  expect(toggle.text()).toBe('Stop')
  const currentMs = getMillisecondsInt(wrapper)
  expect(currentMs > 0).toBe(true)

  clickToggle(wrapper)
  expect(toggle.text()).toBe('Start')

  clickClear(wrapper)
  expect(toggle.text()).toBe('Start')
  expect(getMillisecondsInt(wrapper)).toBe(0)
})

function clickToggle(wrapper) {
  getToggle(wrapper).simulate('click')
}

function clickClear(wrapper) {
  wrapper.find('[data-test="clear"]').simulate('click')
}

function getToggle(wrapper) {
  return wrapper.find('[data-test="toggle"]')
}

function getMilliseconds(wrapper) {
  return wrapper.find('[data-test="ms"]')
}

function getMillisecondsInt(wrapper) {
  const text = getMilliseconds(wrapper).text()
  return Number(text.substring(0, text.length - 2))
}
