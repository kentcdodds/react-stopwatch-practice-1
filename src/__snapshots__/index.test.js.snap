exports[`test initial state 1`] = `
<StopWatch>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }>
    <label
      data-test="ms"
      style={
        Object {
          "display": "block",
          "fontSize": "5em",
        }
      } />
    <button
      data-test="toggle"
      onClick={[Function]}
      style={
        Object {
          "background": "#fff",
          "border": "1px solid #ccc",
          "fontSize": "2em",
          "margin": "0 5px",
          "padding": "15px",
          "width": "200px",
        }
      }>
      Start
    </button>
    <button
      data-test="clear"
      onClick={[Function]}
      style={
        Object {
          "background": "#fff",
          "border": "1px solid #ccc",
          "fontSize": "2em",
          "margin": "0 5px",
          "padding": "15px",
          "width": "200px",
        }
      }>
      Clear
    </button>
  </div>
</StopWatch>
`;
